### PR TITLE
ACMS-1082: Add API modules to ACMS.

### PR DESCRIPTION
**Pull request by modularlab**
**Motivation**
Adds some additional API modules and sets as dependencies of ACMS Headless.

**Proposed changes**
- Adds [OpenAPI UI](https://www.drupal.org/project/openapi_ui) module.
- Adds [ReDoc for OpenAPI UI](https://www.drupal.org/project/openapi_ui_redoc) module.
- Adds [OpenAPI JSON:API](https://www.drupal.org/project/openapi_jsonapi) module.
- Sets ReDoc for OpenAPI UI and OpenAPI JSON:API as composer dependencies both in ACMS and ACMS Headless.
- Sets as installed dependencies in ACMS Headless info yaml.
- 
**Alternatives considered**
N / A

**Testing steps**
- Pull down branch
- Install the ACMS Headless module
- Verify that ReDoc for OpenAPI UI, OpenAPI JSON:API, OpenAPI UI, and any other dependencies are installed along with ACMS Headless.

**Merge requirements**
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by modularlab** [comment]
@msherron, @nagwani, @vishalkhode1

This is ready for review.

File: composer.json
Comment (vishalkhode1) [inline composer.json:61]: @modularlab Sorry, I missed this in earlier review. Can we unpin this module version ?

>If any patches needs to be added, then pin the module version (instead of using min. stability version. Ex: ‘drupal/next:^1.0’ shouldn’t be used instead ‘drupal/next:1.0.0’ should be used). 

By this, what I mean is: If we are adding patch for the module which we've added in composer.json, then we should pin the module version, else it's not needed. We haven't any patches for `drupal/next` module, we've added patch for modules: `drupal/decoupled_router` & `drupal/subrequests`. So, we don't need to pin `drupal/next` module.

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

**Comment by vishalkhode1** [comment]
@modularlab We've updated [feature/bounteous-headless](https://github.com/acquia/acquia_cms/tree/feature/bounteous-headless) branch with latest changes, please resolve the conflicts and rebase your branch with [feature/bounteous-headless](https://github.com/acquia/acquia_cms/tree/feature/bounteous-headless) branch.

File: composer.json
Comment (modularlab) [reply composer.json:61]: No worries.  I've actually addressed this in #1098.  I'll also hold off on fixing any composer conflicts on this PR until #1098 has been merged.

**Comment by modularlab** [comment]
Waiting for #1098 to be merged prior to addressing conflicts on this PR.  Issue with Next.js being pinned has also been resolved via 1098.

**Comment by msherron** [comment]
@modularlab - I merged #1098 for you, you're good to proceed.

**Review by vishalkhode1 — APPROVED** [review]
Looks good to me.

**Pull request merged**
